include/sys/asm.h

```diff
@@ -22,49 +22,84 @@ modified versions thereof.
 extern "C" {
 #endif
 
-#define _MIPS_ISA_MIPS1	1	/* R2/3K */
-#define _MIPS_ISA_MIPS2	2	/* R4K/6K */
-#define _MIPS_ISA_MIPS3	3	/* R4K */
-#define _MIPS_ISA_MIPS4	4	/* TFP */
+#define _MIPS_ISA_MIPS1 1   /* R2/3K */
+#define _MIPS_ISA_MIPS2 2   /* R4K/6K */
+#define _MIPS_ISA_MIPS3 3   /* R4K */
+#define _MIPS_ISA_MIPS4 4   /* TFP */
 
-#define _MIPS_SIM_ABI32		1	/* MIPS MSIG calling convention */
-#define _MIPS_SIM_NABI32	2	/* MIPS new 32-bit abi */
-		/* NABI32 is 64bit calling convention but 32bit type sizes) */
-#define _MIPS_SIM_ABI64		3	/* MIPS 64 calling convention */
+#define _MIPS_SIM_ABI32     1   /* MIPS MSIG calling convention */
+#define _MIPS_SIM_NABI32    2   /* MIPS new 32-bit abi */
+        /* NABI32 is 64bit calling convention but 32bit type sizes) */
+#define _MIPS_SIM_ABI64     3   /* MIPS 64 calling convention */
 
-#define	LEAF(x)						\
-	.globl	x;					\
-	.ent	x,0;					\
-x:;							\
-	.frame	sp,0,ra
 
-#define	XLEAF(x)					\
-	.global x;
+#define LEAF(x)             \
+    .globl  x              ;\
+    .align  2              ;\
+    .ent    x,0            ;\
+    x:                     ;\
+    .frame  sp,0,ra
+    //.type   x, @function   ;\
 
-#define	END(proc)					\
-	.end	proc
+#if defined(BBPLAYER) || defined(__sgi)
+#define XLEAF(x)    \
+    .globl  x      ;\
+    x:
```

Comment:
```suggestion
    .globl  x      ;\
    .aent   x,0;   ;\
    x:
```
Does egcs allow this? (this is needed for irix which you've added the define for)

Reply:
`aent` is theoretically supported by this version of gas but for some reason it claims it's unsupported. I guess I'll need to add another version of the macro for irix unless that's figured out